File: frontend/src/game/objects/bomb.ts
import { COUNTDOWN_MS } from "../gameConfig";
import { ANIMS } from "../sprite_animations/animations";
import { ManSpriteKey, SpriteKey } from "../sprite_animations/sprite";
import { MapIndex } from "../types";
import { BaseObj } from "./base";

export class BombObj extends BaseObj {
    remainingMs: number = COUNTDOWN_MS.bomb
    constructor(
        scene: Phaser.Scene,
        index: MapIndex,
        public power: number,
        public manSpriteKey: ManSpriteKey
    ) {
        super(scene, index, "items", 0, "bomb");
        this.sprite.anims.play(ANIMS.item.bomb)
    }
    triggerExplode() {
        //TODO
    }
}

File: backend/src/game/objects/bomb.rs
use crate::game::config::BOMB_COUNTDOWN_MS;
use crate::game::types::{ManSpriteKey, MapIndex, ObjType};
use super::base::BaseObj;

pub struct BombObj {
    pub base: BaseObj,
    pub remaining_ms: u32,
    pub power: u32,
    pub man_sprite_key: ManSpriteKey,
}

impl BombObj {
    pub fn new(index: &MapIndex, power: u32, man_sprite_key: ManSpriteKey) -> Self {
        Self {
            base: BaseObj::new(index, ObjType::Bomb),
            remaining_ms: BOMB_COUNTDOWN_MS,
            power,
            man_sprite_key,
        }
    }
}
